core/controllers/access_validators.py

```diff
@@ -78,6 +78,35 @@ def get(self) -> None:
             raise self.NotFoundException
 
 
+class SubtopicViewerPageAccessValidator(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """ Validates access to the Subtopic Viewer Page """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'classroom_url_fragment': constants.SCHEMA_FOR_CLASSROOM_URL_FRAGMENTS,
+        'topic_url_fragment': constants.SCHEMA_FOR_TOPIC_URL_FRAGMENTS,
+        'subtopic_url_fragment': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.VALID_URL_FRAGMENT_REGEX
+                }, {
+                    'id': 'has_length_at_most',
+                    'max_value': constants.MAX_CHARS_IN_SUBTOPIC_URL_FRAGMENT
+                }]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {'GET': {}}
+
+    @acl_decorators.can_access_subtopic_viewer_page
+    def get(self, *args: str) -> None:
+        """ Handles GET requests """
+        pass
```

Comment:
this isn't calling self.render_json(); could that be the issue?

Reply:
Actually in the SubtopicViewer, there is a data handler calling a render_json

Reply:
Sorry, I don't understand. All GET JSON handlers end with self.render_json({something}) -- why doesn't this one do so?